File: backend/src/main/java/com/lawencon/ticketms/service/TicketPriorityService.java
package com.lawencon.ticketms.service;

import com.lawencon.ticketms.dto.ticketpriority.TicketPriorityListResDto;
import com.lawencon.ticketms.dto.ticketpriority.TicketPriorityResDto;

public interface TicketPriorityService {
	TicketPriorityListResDto getAll();
	TicketPriorityResDto getById(Long id);
	TicketPriorityResDto getByCode(String code);
}

File: frontend/src/app/service/ticket-priority.service.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Observable } from "rxjs";
import { BASE_URL } from "../constant/base.url";
import { TicketPriorityListResDto } from "../dto/ticket-priority/ticket-priority.list.res.dto";

@Injectable({
    providedIn : 'root'
})
export class TicketPriorityService {

    constructor(private http : HttpClient) {}

    getAll() : Observable<TicketPriorityListResDto> {
        return this.http.get<TicketPriorityListResDto>(`${BASE_URL.LOCALHOST}/ticket_priority`)
    }

}